compio-buf/src/io_buf.rs

```diff
@@ -566,7 +533,8 @@ impl<T: IoBufMut, const N: usize> IoIndexedBufMut for arrayvec::ArrayVec<T, N> {
 
 /// A helper trait for `set_len` like methods.
 pub trait SetBufInit {
-    /// Updates the number of initialized bytes.
+    /// Set the buffer length. If `len` is less than the current length, nothing
+    /// will happen.
```

Comment:
Change to "nothing should happen" because the behavior actually depend on the implementor. Another way to enforce this is to seal the trait.